frontend/utils/equipment.ts

```diff
@@ -0,0 +1,44 @@
+import { clamp } from "remeda";
+
+
+
+export interface IMockEquipment {
+  name: string;
+  id: number;
+  availability: number;
+  quality: number;
+  performance: number;
+  get oee(): number;
+}
+
+export interface Metric {
+  label: string;
+  value: number;
+  displayValue: number;
+}
+
+export type MetricKey = "availability" | "quality" | "performance" | "oee";
+
+export class MockEquipment {
+  data: IMockEquipment;
+
+  constructor(equipment: IMockEquipment) {
```

Comment:
Passing in an object to a constructor is a bit of an anti-pattern. I'd recommend returning to the previous structure of mock equipment to avoid over-engineering, but I won't press that in this PR.